src/midi/midi_fluidsynth.cpp

```diff
@@ -134,39 +134,39 @@ std::tuple<std::string, int> parse_sf_pref(const std::string &line,
 
 #if defined(WIN32)
 
-static std::deque<std::string> get_data_dirs()
+static std::deque<std_fs::path> get_data_dirs()
 {
 	return {
-	        CROSS_GetPlatformConfigDir() + "soundfonts\\",
+	        std_fs::path(CROSS_GetPlatformConfigDir()) / "soundfonts",
 	        "C:\\soundfonts\\",
 	};
 }
 
 #elif defined(MACOSX)
 
-static std::deque<std::string> get_data_dirs()
+static std::deque<std_fs::path> get_data_dirs()
 {
 	return {
-	        CROSS_GetPlatformConfigDir() + "soundfonts/",
-	        CROSS_ResolveHome("~/Library/Audio/Sounds/Banks/"),
+	        std_fs::path(CROSS_GetPlatformConfigDir()) / "soundfonts",
+	        std_fs::path(CROSS_ResolveHome("~/Library/Audio/Sounds/Banks")),
```

Comment:
Looking at CROSS_GetPlatformConfigDir(), maybe it can cache and return a std::fs::path object (to avoid repeated wrapper construction from strings)?

It's not used in much code, so ripples should be pretty minor: 

```
git grep  -l CROSS_GetPlatformConfigDir

gui/sdlmain.cpp
midi/midi_fluidsynth.cpp                             midi/midi_mt32.cpp
misc/cross.cpp                                       misc/support.cpp
```

Reply:
I was wondering if I should just clean it up as well :smile:. I'll try to do it, and maybe even limit the usage of `Cross::GetPlatformConfigDir` at the same time.

Reply:
Very nice :-)
Approved either approach; please feel free to merge when passing!

Reply:
Of course the refactor turned out to be non-trivial... Decorated code with asserts and deprecations until it's done (which means CI will fail on deprecations for now). Setting PR to draft for now.

Reply:
Generally, I don't care optimising code than only gets called a handful of times and optimise for simplicity and readability. Why does it matter that these objects are not cached We're talking about less than 10 calls perhaps when running the program, not millions.

Reply:
@johnnovak , fully agree that caching isn't needed.

Reply:
I don't care about optimisation in this case either (it's just a happy accident) - but I do care about deprecating/removing functions that return paths as strings (for code readability).

Old DOSBox code has this annoying  property that it (usually, not always) expects directory separator to be returned from functions returning directory paths... Which leads to confusing code, as readers can't easily distinguish if code is joining paths or appending suffix to path.

Reply:
Ugh, this task grew a bit due to all places I needed to touch to remove old `CROSS_GetPlatformConfigDir`; I think reviewing commit-by-commit might be easier.

@kcgen I will submit changes for `CROSS_ResolveHome` in another PR, I think this one is big enough. Ready for re-review 😄.